frontend/src/components/Search/SearchForm/SearchForm.tsx

```diff
@@ -240,46 +243,65 @@ export const Search: React.FC<{
 	const { project_id } = useParams()
 	const containerRef = useRef<HTMLDivElement | null>(null)
 	const inputRef = useRef<HTMLInputElement | null>(null)
+	const [keys, setKeys] = useState<Keys | undefined>()
+	const [values, setValues] = useState<string[] | undefined>()
 	const comboboxStore = useComboboxStore({
 		defaultValue: query ?? '',
 	})
-	const [getKeys, { data: keysData, loading: keysLoading }] =
-		fetchKeysLazyQuery()
-	const [getKeyValues, { data, loading: valuesLoading }] =
-		fetchValuesLazyQuery()
+	const [getKeys, { loading: keysLoading }] = fetchKeysLazyQuery()
+	const [getKeyValues, { loading: valuesLoading }] = fetchValuesLazyQuery()
 	const [cursorIndex, setCursorIndex] = useState(0)
+	const [isPending, startTransition] = React.useTransition()
 
 	const { queryParts, tokens } = parseSearch(query)
 	const tokenGroups = buildTokenGroups(tokens, queryParts, query)
 	const activePart = getActivePart(cursorIndex, queryParts)
-	const debouncedKeyValue = useDebouncedValue<string>(activePart.value)
+	const { debouncedValue, setDebouncedValue } = useDebounce<string>(
```

Comment:
Not in this PR, but should we be using `useDebounce` > `useDebouncedValue` in all places?

Reply:
Not something we need to do everywhere, I just wanted to be able to update the debounced value independently in that one case where someone is transitioning to a new query filter. `useDebouncedValue` didn't return a setter to do that, so I created my own hook here. Sorry if you think that will be a source of confusion.

Reply:
That make sense to me! Feels like we should move towards one, but I don't feel strongly about the urgency